src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -0,0 +1,53 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+public class ShooterSubsystem extends SubsystemBase {
+
+  private CANSparkMax shooterMotor;
+  boolean readyToFire = true; // Take in readiness state from ShooterTrigger?
+
+  public ShooterSubsystem() {
+    shooterMotor =
+        new CANSparkMax(Constants.ShooterConstants.SHOOTER_MOTOR_ID, MotorType.kBrushless);
+    shooterMotor.restoreFactoryDefaults();
+    shooterMotor.setSmartCurrentLimit(40);
+  }
+
+  public Command shooterFire(double distance) {
```

Comment:
Same for function parameters; please make them all `final`.

Reply:
Resolved